lib/graph_metrics_unittest.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
+# Copyright 2014 Google.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the graph_metrics module"""
+
+import encoder
+import fileset_picker
+import graph_metrics
+import optimizer
+import score_tools
+import test_tools
+import unittest
+
+class FakeCodec1(encoder.Codec):
+  def __init__(self, name='mock1'):
+    super(FakeCodec1, self).__init__(name)
+    self.option_set = encoder.OptionSet()
+
+  def StartEncoder(self, context):
+    return encoder.Encoder(context, encoder.OptionValueSet(self.option_set, ''))
+
+
+class FakeCodec2(FakeCodec1):
+  def __init__(self):
+    super(FakeCodec2, self).__init__('mock2')
+    self.name = 'mock2'
```

Comment:
Is this really needed? Doesn't the previous line already set self.name via encoder.Codec.__init__(self, name)?